--- origsrc/gimp-2.4.6/configure.in	2008-05-23 04:41:20.000000000 -0500
+++ src/gimp-2.4.6/configure.in	2008-06-01 01:39:39.609375000 -0500
@@ -1593,15 +1593,22 @@
   dnl This might be able to be figured out from distutils, but it's
   dnl not documented so we won't rely on internal implementation
   PYLINK_LIBS=
-  if test "x$platform_win32" = "xyes"; then
-    PYBIN_PATH="$py_exec_prefix\pythonw.exe"
-    PYTHON_INCLUDES=`$echo "$PYTHON_INCLUDES" | sed -e 's/\\\\/\\//g'`
-    py_exec_prefix=`echo "$py_exec_prefix" | sed -e 's/\\\\/\\//g'`
-    pylibversion=`echo $PYTHON_VERSION | sed -e 's/\\.//'`
-    PYLINK_LIBS="-L${py_exec_prefix}/libs -lpython${pylibversion}"
-  else
-    PYBIN_PATH="$PYTHON"
-  fi
+  case "$target_or_host" in
+    *-*-cygwin*)
+      PYBIN_PATH="$PYTHON"
+      PYLINK_LIBS="-L${py_exec_prefix}/lib/python${PYTHON_VERSION}/config -lpython${PYTHON_VERSION}"
+      ;;
+    *-*-mingw*)
+      PYBIN_PATH="$py_exec_prefix\pythonw.exe"
+      PYTHON_INCLUDES=`$echo "$PYTHON_INCLUDES" | sed -e 's/\\\\/\\//g'`
+      py_exec_prefix=`echo "$py_exec_prefix" | sed -e 's/\\\\/\\//g'`
+      pylibversion=`echo $PYTHON_VERSION | sed -e 's/\\.//'`
+      PYLINK_LIBS="-L${py_exec_prefix}/libs -lpython${pylibversion}"
+      ;;
+    *)
+      PYBIN_PATH="$PYTHON"
+      ;;
+  esac
   AC_SUBST(PYLINK_LIBS)
   AC_SUBST(PYBIN_PATH)
 
--- origsrc/gimp-2.4.6/plug-ins/pygimp/Makefile.am	2007-11-20 03:33:14.000000000 -0600
+++ src/gimp-2.4.6/plug-ins/pygimp/Makefile.am	2008-06-01 01:39:39.640625000 -0500
@@ -13,9 +13,6 @@
 
 if PLATFORM_WIN32
 no_undefined = -no-undefined
-path_separator = ;
-else
-path_separator = :
 endif
 
 AM_CFLAGS = $(PYGIMP_EXTRA_CFLAGS)
@@ -139,7 +136,7 @@
 
 install-env-file:
 	$(mkinstalldirs) '$(DESTDIR)$(pyenvdir)'
-	echo '$(path_separator) PYTHONPATH=$${gimp_plug_in_dir}/$(pygimpbase)' > '$(DESTDIR)$(pyenvfile)'
+	echo '$(PATHSEP) PYTHONPATH=$${gimp_plug_in_dir}/$(pygimpbase)' > '$(DESTDIR)$(pyenvfile)'
 
 install-interp-file:
 	$(mkinstalldirs) '$(DESTDIR)$(pyinterpdir)'
