--- origsrc/gimp-2.8.0/configure.ac
+++ src/gimp-2.8.0/configure.ac
@@ -727,7 +727,7 @@
 esac
 
 
-if test "x$platform_win32" = "xyes"; then
+if test "x$os_win32" = "xyes"; then
   shmtype=win32
 elif test "x$shmtype" = "xsysv"; then
   AC_CHECK_HEADER(sys/ipc.h,
@@ -810,7 +810,7 @@
 AC_MSG_CHECKING([whether symbols are prefixed])
 
 case "$host_os" in
-  mingw*)
+  mingw* | cygwin*)
     case "$host_cpu" in 
 	  x86_64) SYMPREFIX='' ;;
 	  *) SYMPREFIX='_' ;;
--- origsrc/gimp-2.6.12/app/plug-in/gimpplugin.c	2012-01-31 07:50:14.000000000 -0600
+++ src/gimp-2.6.12/app/plug-in/gimpplugin.c	2012-02-06 16:41:50.741351900 -0600
@@ -48,7 +48,7 @@
 
 #include <glib-object.h>
 
-#if defined(G_OS_WIN32) || defined(G_WITH_CYGWIN)
+#if defined(G_OS_WIN32)
 
 #define STRICT
 #include <windows.h>
@@ -71,7 +71,7 @@
 #define _O_BINARY 0x0200  /* binary file */
 #endif
 
-#endif /* G_OS_WIN32 || G_WITH_CYGWIN */
+#endif /* G_OS_WIN32 */
 
 #include "libgimpbase/gimpbase.h"
 #include "libgimpbase/gimpprotocol.h"
@@ -112,7 +112,7 @@ static gboolean   gimp_plug_in_recv_mess
                                               GIOCondition  cond,
                                               gpointer      data);
 
-#if !defined(G_OS_WIN32) && !defined (G_WITH_CYGWIN)
+#if !defined(G_OS_WIN32)
 static void       gimp_plug_in_prep_for_exec (gpointer      data);
 #else
 #define           gimp_plug_in_prep_for_exec  NULL
@@ -252,14 +252,6 @@ gimp_plug_in_open (GimpPlugIn         *p
       return FALSE;
     }
 
-#if defined(G_WITH_CYGWIN)
-  /* Set to binary mode */
-  setmode (my_read[0], _O_BINARY);
-  setmode (my_write[0], _O_BINARY);
-  setmode (my_read[1], _O_BINARY);
-  setmode (my_write[1], _O_BINARY);
-#endif
-
 #ifdef G_OS_WIN32
   /* Prevent the plug-in from inheriting our ends of the pipes */
   SetHandleInformation ((HANDLE) _get_osfhandle (my_read[0]), HANDLE_FLAG_INHERIT, 0);
@@ -738,7 +730,7 @@ gimp_plug_in_flush (GIOChannel *channel,
   return TRUE;
 }
 
-#if !defined(G_OS_WIN32) && !defined (G_WITH_CYGWIN)
+#if !defined(G_OS_WIN32)
 
 static void
 gimp_plug_in_prep_for_exec (gpointer data)
--- origsrc/gimp-2.6.12/app/plug-in/gimppluginshm.c	2012-01-31 07:50:15.000000000 -0600
+++ src/gimp-2.6.12/app/plug-in/gimppluginshm.c	2012-02-06 16:44:13.541985300 -0600
@@ -47,7 +47,7 @@
 
 #include <glib-object.h>
 
-#if defined(G_OS_WIN32) || defined(G_WITH_CYGWIN)
+#if defined(G_OS_WIN32)
 
 #define STRICT
 #include <windows.h>
@@ -60,7 +60,7 @@
 
 #define USE_WIN32_SHM 1
 
-#endif /* G_OS_WIN32 || G_WITH_CYGWIN */
+#endif /* G_OS_WIN32 */
 
 #include "plug-in-types.h"
 
--- origsrc/gimp-2.6.12/libgimp/gimp.c	2012-01-31 07:50:15.000000000 -0600
+++ src/gimp-2.6.12/libgimp/gimp.c	2012-02-06 16:44:27.700283200 -0600
@@ -83,7 +83,7 @@
 
 #endif /* USE_POSIX_SHM */
 
-#if defined(G_OS_WIN32) || defined(G_WITH_CYGWIN)
+#if defined(G_OS_WIN32)
 #  define STRICT
 #  define _WIN32_WINNT 0x0601
 #  include <windows.h>
